modules/nextflow/src/main/groovy/nextflow/cli/CmdRun.groovy

```diff
@@ -355,7 +356,8 @@ class CmdRun extends CmdBase implements HubOptions {
 
         final isTowerEnabled = config.navigate('tower.enabled') as Boolean
         final isDataEnabled = config.navigate("lineage.enabled") as Boolean
-        if( isTowerEnabled || isDataEnabled || log.isTraceEnabled() )
+
+        if( isTowerEnabled || isDataEnabled || LoggerFactory.getLogger(nextflow.config.ConfigBuilder).isTraceEnabled() )
```

Comment:
I think we can just remove these conditions and resolve the config always

Reply:
I never understood why we resolve it here anyway, when it is already resolved here

https://github.com/nextflow-io/nextflow/blob/2c9dd9b18a179b28a46ed998c165acc47107f74d/modules/nextflow/src/main/groovy/nextflow/cli/CmdRun.groovy#L329-L333

Reply:
Because this it's done to render it as a string using 

https://github.com/nextflow-io/nextflow/blob/e4af2308d5c1a4fd16438005565fefcf07891af7/modules/nextflow/src/main/groovy/nextflow/config/ConfigBuilder.groovy#L908-L909

maybe it can be done reusing the config object already build, but never looked into that